runtime_scan/pkg/provider/docker/client.go

```diff
@@ -0,0 +1,464 @@
+	}
+
+	return &Client{
+		dockerClient: dockerClient,
+	}, nil
+}
+
+func (c *Client) Kind() models.CloudProvider {
+	return models.Docker
+}
+
+func (c *Client) DiscoverAssets(ctx context.Context) ([]models.AssetType, error) {
+	// Get image assets
+	imageAssets, err := c.getImageAssets(ctx)
+	if err != nil {
+		return nil, provider.FatalErrorf("failed to get images. Provider=%s: %w", models.Docker, err)
+	}
+
+	// Get container assets
+	containerAssets, err := c.getContainerAssets(ctx)
+	if err != nil {
+		return nil, provider.FatalErrorf("failed to get containers. Provider=%s: %w", models.Docker, err)
+	}
+
+	// Combine assets
+	assets := append(imageAssets, containerAssets...)
+
+	return assets, nil
+}
+
+func (c *Client) RunAssetScan(ctx context.Context, config *provider.ScanJobConfig) error {
+	assetVolume, err := c.prepareScanAssetVolume(ctx, config)
+	if err != nil {
+		return provider.FatalErrorf("failed to prepare scan volume. Provider=%s: %w", models.Docker, err)
+	}
+
+	networkID, err := c.createScanNetwork(ctx)
+	if err != nil {
+		return provider.FatalErrorf("failed to prepare scan network. Provider=%s: %w", models.Docker, err)
+	}
+
+	containerID, err := c.createScanContainer(ctx, assetVolume, networkID, config)
+	if err != nil {
+		return provider.FatalErrorf("failed to create scan container. Provider=%s: %w", models.Docker, err)
+	}
+
+	err = c.dockerClient.ContainerStart(ctx, containerID, types.ContainerStartOptions{})
+	if err != nil {
+		return provider.FatalErrorf("failed to start scan container. Provider=%s: %w", models.Docker, err)
+	}
+
+	return nil
+}
+
+func (c *Client) RemoveAssetScan(ctx context.Context, config *provider.ScanJobConfig) error {
+	containerID, err := c.getContainerIDFromName(ctx, config.AssetScanID)
+	if err != nil {
+		return provider.FatalErrorf("failed to get scan container id. Provider=%s: %w", models.Docker, err)
+	}
+	err = c.dockerClient.ContainerRemove(ctx, containerID, types.ContainerRemoveOptions{Force: true})
+	if err != nil {
+		return provider.FatalErrorf("failed to remove scan container. Provider=%s: %w", models.Docker, err)
+	}
+
+	scanConfigFileName := getScanConfigFilePath(config)
+	err = os.Remove(scanConfigFileName)
+	if err != nil {
+		return provider.FatalErrorf("failed to remove scan config file. Provider=%s: %w", models.Docker, err)
+	}
+
+	err = c.dockerClient.VolumeRemove(ctx, config.AssetScanID, true)
+	if err != nil {
+		return provider.FatalErrorf("failed to remove volume. Provider=%s: %w", models.Docker, err)
+	}
+
+	networkID, err := c.getNetworkIDFromName(ctx, config.AssetScanID)
+	if err != nil {
+		return provider.FatalErrorf("failed to get scan network id. Provider=%s: %w", models.Docker, err)
+	}
+	err = c.dockerClient.NetworkRemove(ctx, networkID)
+	if err != nil {
+		return provider.FatalErrorf("failed to remove scan network. Provider=%s: %w", models.Docker, err)
+	}
```

Comment:
This will create race issues since the network is static and is used by all runners, plus it will fail every time there's another container attached to it. I think it's fine to have this network present without having to remove it once created. We can remove this cleanup action.

Reply:
removed, thanks!